### PR TITLE
feat: support vue sfc

### LINKED ISSUE #49: Not sorting imports in .vue files
Hi, I have a following config:

```json
{
  "importOrder": [
    "^vue/(.*)$",
    "^[^@v](.*)$",
    "^@(.*)$",
    "^[./]"
  ],
  "importOrderSeparation": true,
}

```

And an `App.vue` file (already formatted with prettier):

```vue
<script>
import Vue from 'vue'
import TheNavBar from '@/components/TheNavBar'
import dayjs from 'dayjs'
</script>
```

When I copy its content to `.ts` file, it gets formatted as expected:
```typescript
import Vue from 'vue'

import dayjs from 'dayjs'

import TheNavBar from '@/components/TheNavBar'

```

Is it possible to somehow enable `.vue` files support?

### PR DESCRIPTION
**Pull request by blake-newman**
resolves #49

### CODE REVIEW

**Comment by byara** [comment]
Hey @blake-newman thank you for your contribution. We will take a look 👍

**Comment by blake-newman** [comment]
I'm thinking may be best to abstract this in a better way to avoid conditions. Can have a separate processor. Going to push changes up shortly

File: package.json [hunk 1]
```diff
@@ -38,17 +38,19 @@
...
     "@types/jest": "26.0.20",
-    "@types/node": "14.14.34",
     "@types/lodash": "4.14.168",
+    "@types/node": "14.14.34",
     "jest": "26.6.3",
     "prettier": "2.3.1",
     "ts-jest": "26.5.3",
-    "typescript": "4.2.3"
+    "typescript": "4.2.3",
+    "vue": "^3.2.40"
```
Comment (byara) [inline package.json:53]: Is `vue` really needed?

File: src/preprocessor.ts [hunk 2]
```diff
@@ -1,13 +1,29 @@
...
+    const content = (descriptor.script ?? descriptor.scriptSetup)?.content ?? code
+
+    return code.replace(
+        content,
+        `\n${process(content, options)}\n`,
+    );
+
+}
+
+function process(code: string, options: PrettierOptions) { {
```
Comment (byara) [inline src/preprocessor.ts:26]: I would suggest to make a preprocessor folder and separate them in different files. I would say `process` also can go in a separate file. Maybe `preprocessor` can also be renamed to something like `defaultPreprocessor`? What do you think?

**Review by byara — COMMENTED** [review]
Thank you for your contribution 🚀 
I left some comments. It would be great if we could address them.

**Comment by ayusharma** [comment]
FYI: I will open and re-opening the branch to fix the Github CI issue.

File: package.json (hunk 1 above)
Comment (blake-newman) [reply package.json]: No; i've removed :+1:

File: src/preprocessor.ts (hunk 2 above)
Comment (blake-newman) [reply src/preprocessor.ts]: Yeah i've done the changes as requested. I'm happy with whatever you feel is best. Your repository after all :)

**Comment by ayusharma** [comment]
> @ayusharma i fixed the issue with the snapshot and also updated the README to reflect support for vue.

Thank you so much, I will check it today.

**Review by ayusharma — APPROVED** [review]
The PR looks good to me. 🎉  Very nice work and thank you so much for the contribution @blake-newman ❤️  

I would suggest releasing this change with a new major version v4.0.0. What do you think @blake-newman and @byara ?

**Comment by blake-newman** [comment]
Personally; i'd suggest even a patch/minor is fine as non-breaking and no changes in the configuration are needed from a user's perspective.

**Comment by ayusharma** [comment]
> Personally; i'd suggest even a patch/minor is fine as non-breaking and no changes in the configuration are needed from a user's perspective.

Okay. I will release it with v3.4.0.

**Review by byara — APPROVED** [review]

**Comment by ayusharma** [comment]
Released in v3.4.0  🎉  Thank you again @blake-newman 😃

**Comment by ricardocosta** [comment]
@blake-newman @ayusharma Hello! First of all, thanks for adding the support to Vue files. I wanted to run something by you regarding this PR:

With Vue support, a dependency on `@vue/compiler-sfc` was added, which brings along quite a few other dependencies:
```
"@vue/compiler-sfc@npm:^3.2.40":
  version: 3.2.41
  resolution: "@vue/compiler-sfc@npm:3.2.41"
  dependencies:
    "@babel/parser": ^7.16.4
    "@vue/compiler-core": 3.2.41
    "@vue/compiler-dom": 3.2.41
    "@vue/compiler-ssr": 3.2.41
    "@vue/reactivity-transform": 3.2.41
    "@vue/shared": 3.2.41
    estree-walker: ^2.0.2
    magic-string: ^0.25.7
    postcss: ^8.1.10
    source-map: ^0.6.1
```

Do you see a way of supporting Vue, but keeping the dependency in a non-required way (e.g. `peerDependency`)? Non-Vue projects that use this plugin will now have that dependency installed for no reason.

**Comment by blake-newman** [comment]
It may be possible to make it a peer dependency, may need to do inline require in preproccesor layer so lazy load the Vue compiler. 

I'll take a look soon 👍

**Pull request merged**